src/argilla/server/contexts/datasets.py

```diff
@@ -256,7 +257,10 @@ async def get_record_by_id(
 ) -> Union[Record, None]:
     query = select(Record).filter_by(id=record_id)
     if with_dataset:
-        query = query.options(selectinload(Record.dataset).selectinload(Dataset.questions))
+        query = query.options(
+            selectinload(Record.dataset).selectinload(Dataset.questions),
+            selectinload(Record.dataset).selectinload(Dataset.metadata_properties),
```

Comment:
Do we always need `questions` and `metadata_properties` when getting a record via `get_record_by_id`?

Reply:
probably not, I would change this